pinot-core/src/main/java/org/apache/pinot/core/common/datatable/DataTableUtils.java

```diff
@@ -71,10 +71,12 @@ public static int computeColumnOffsets(DataSchema dataSchema, int[] columnOffset
         case LONG:
           rowSizeInBytes += 8;
           break;
-        // TODO: fix float size (should be 4).
-        // For backward compatible, DON'T CHANGE.
         case FLOAT:
-          rowSizeInBytes += 8;
+          if (dataTableVersion >= 4) {
```

Comment:
During upgrade broker should still be able to handle receiving FLOAT as 8 bytes since broker is upgraded first and there will be new broker, old server for some point of time.

Reply:
yeah it should be able to handle this since the dataTableVersion is provided by the initial 4 bytes of the byte buffer. it will adjust the float type size accordingly.